### PR TITLE
Also remove const-hack for abs

### LINKED ISSUE #67842: `abs` function still not cleaned up from const_hack
#67657 seems to have missed the `abs` function which is still hacky.

cc @oli-obk @jumbatm

This issue has been assigned to @jumbatm via [this comment](https://github.com/rust-lang/rust/issues/67842#issuecomment-570718763).

### PR DESCRIPTION
**Pull request by jumbatm**
Closes #67842.

r? @oli-obk

### CODE REVIEW

**Comment by oli-obk** [comment]
cc @RalfJung @Centril 

@bors r+

Thanks!

File: src/libcore/num/mod.rs [hunk 1]
```diff
@@ -1997,27 +1997,15 @@ $EndFeature, "
 ```"),
             #[stable(feature = "rust1", since = "1.0.0")]
             #[rustc_const_stable(feature = "const_int_methods", since = "1.32.0")]
+            #[allow_internal_unstable(const_if_match)]
             #[inline]
             #[rustc_inherit_overflow_checks]
             pub const fn abs(self) -> Self {
-                // Note that the #[inline] above means that the overflow
-                // semantics of the subtraction depend on the crate we're being
-                // inlined into.
```
Comment (petrochenkov) [inline src/libcore/num/mod.rs]: This comment shouldn't be lost.
(See the original version of this code before the const hack was introduced - https://github.com/rust-lang/rust/commit/39260d9016db6925414541ad2dce6502dded3a64.)

Comment (jumbatm) [reply src/libcore/num/mod.rs]: Understood. Re-added the comment now.

**Comment by tspiteri** [comment]
@jumbatm Why was this PR closed? It seems that while the first commit was merged in the "Rollup of 8 pull requests", the second commit was not merged.

**Comment by jumbatm** [comment]
Ah, I see. It closed when I pushed a delete fork to my branch, sorry!

**Comment by tspiteri** [comment]
@oli-obk Does this PR need to be reopened and r+ed once again?

**Comment by oli-obk** [comment]
Please rebase over master so we see what is unmerged so far

**Comment by jumbatm** [comment]
Done

**Pull request merged**